src/hooks/useExistingRooms.ts

```diff
@@ -0,0 +1,106 @@
+import { useEffect } from 'react'
+import { StreamOperation, StreamrClient } from 'streamr-client'
+import { ActionType, useDispatch, useStore } from '../components/Store'
+import { StorageKey } from '../utils/types'
+import useInviter from './useInviter'
+import intersection from 'lodash/intersection'
+
+const ROOM_PREFIX = 'streamr-chat/room'
+
+export default function useExistingRooms() {
+    const {
+        account,
+        session: { wallet },
+        ethereumProvider,
+    } = useStore()
+
+    const dispatch = useDispatch()
+
+    const invite = useInviter()
+
+    const sessionAccount = wallet?.address
+
+    useEffect(() => {
+        if (!ethereumProvider || !account || !sessionAccount) {
+            return
+        }
+
+        const localRoomIds =
+            ((): string[] | undefined => {
+                try {
+                    return JSON.parse(
+                        localStorage.getItem(StorageKey.RoomIds) || '{}'
+                    )[account]
+                } catch (e) {}
+            })() || []
+
+        dispatch({
+            type: ActionType.SetRoomIds,
+            payload: localRoomIds,
+        })
+
+        // Create a metamask/streamr-client instance to fetch parent identity's streams
+        const providerClient = new StreamrClient({
+            auth: {
+                ethereum: ethereumProvider as any,
+            },
+        })
+
+        async function fn() {
+            const remoteRoomIds: string[] = []
+
+            const streams = await providerClient.listStreams({
+                search: ROOM_PREFIX,
+                operation: 'stream_share' as StreamOperation,
```

Comment:
Not sure which operation to look for. `subscribe` means we can read the chat, but we also want to share it. It used to be `stream_subscribe`. @bearni95 thoughts?